web/src/components/sign-up/sign-up.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import StateTree from '../../stores/tree';
+import API from '../../services/api';
+import { connect } from 'react-redux';
+import { CheckIcon, OldPlayIcon, CautionIcon } from '../ui/icons';
+
+import './sign-up.css';
+
+interface PropsFromState {
+  api: API;
+}
+
+interface State {
+  email: string;
+  buttonStyle: string;
+  buttonHtml: any;
+}
+
+class SignUp extends React.Component<PropsFromState, State> {
+  constructor(props: PropsFromState, context: any) {
+    super(props, context);
+    const { api } = this.props;
+    this.state = {
+      email: '',
+      buttonStyle: 'submit-button',
+      buttonHtml: <OldPlayIcon className="icon" />,
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event: any) {
+    this.setState({ email: event.target.value });
+    if (this.state.buttonStyle != 'submit-button') {
+      this.setState({ buttonStyle: 'submit-button' });
+      this.setState({ buttonHtml: <OldPlayIcon className="icon" /> });
```

Comment:
It's a bit of an anti-pattern in React to have components in the state and I don't think it's necessary in this case. We could remodel the state, so that depending on the status of the form (e.g. initial/error/submitted), it renders different stuff. That'd also allow us to reduce the state properties down to 2.